akka-stream/src/main/scala/akka/stream/impl/fusing/ActorGraphInterpreter.scala

```diff
@@ -510,6 +518,20 @@ private[stream] class ActorGraphInterpreter(_initial: GraphInterpreterShell) ext
         false
     }
 
+  //this limits number of messages that can be processed synchronously during one actor receive.
+  private val shortCircuitLimit: Int = _initial.mat.settings.syncProcessingLimit
+  private var shortCircuitBuffer: ByteBuffer[BoundaryEvent] = _
+  private var currentLimit: Int = shortCircuitLimit
```

Comment:
Ok, will change it to FixedSizeBuffer. Initial intention was to abstract from implementation (as it's enough to have just ByteBuffer methods) - but as soon as it's internal thing it's not so important